### PR TITLE
Hitting `/Playback/Record` now updates context directory

### PR DESCRIPTION
**Pull request by scbedd**
Enabling the `manual` side of restoring. Automatic already works due to processing of `x-recording-assets-file` when sent alongside the `x-recording-file`.

### CODE REVIEW

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy.Tests/IntegrationTests/GitStoreIntegrationRestoreTests.cs [hunk 1]
```diff
@@ -190,5 +190,42 @@ public async Task Scenario3(string inputJson)
...
+            var currentDirectory = Environment.CurrentDirectory;
+            var recordingHandler = new RecordingHandler(currentDirectory, _defaultStore);
+            
+            try
+            {
+                await recordingHandler.Restore(pathToAssets);
+
+                var result = (await _defaultStore.ParseConfigurationFile(pathToAssets)).AssetsRepoLocation;
+
+                Assert.Equal(result, recordingHandler.ContextDirectory);
```
Comment (JimSuplizio) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy.Tests/IntegrationTests/GitStoreIntegrationRestoreTests.cs:223]: We should also assert the file count and file contents, you should be able to copy those directly from Scenario3.

**Review by JimSuplizio — APPROVED** [review]
pre-approved with one minor comment

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy.Tests/IntegrationTests/GitStoreIntegrationRestoreTests.cs (hunk 1 above)
Comment (scbedd) [reply tools/test-proxy/Azure.Sdk.Tools.TestProxy.Tests/IntegrationTests/GitStoreIntegrationRestoreTests.cs:223]: Agree.

**Pull request merged**